File: 3961-maximize-sum-of-device-ratings/3961-maximize-sum-of-device-ratings.swift
class Solution {
    func maxRatings(_ units: [[Int]]) -> Int {
        let m = units.count
        let n = units[0].count

        if n == 1 {
            var sum = 0
            for i in 0..<m {
                sum += units[i][0]
            }
            return sum
        }

        var total = 0
        var miniF = 1_000_000_000
        var miniS = 1_000_000_000

        for i in 0..<m {
            var firMin = 1_000_000_000
            var secMin = 1_000_000_000

            for value in units[i] {
                if value < firMin {
                    secMin = firMin
                    firMin = value
                } else if value < secMin {
                    secMin = value
                }
            }

            total += secMin
            miniF = min(miniF, firMin)
            miniS = min(miniS, secMin)
        }

        return total - miniS + miniF
    }
}

File: 3961-maximize-sum-of-device-ratings/3961-maximize-sum-of-device-ratings.cpp
class Solution {
public:
    long long maxRatings(vector<vector<int>>& units) {
        int m = units.size() ;
        int n = units[0].size() ;

        if(n == 1){
            long long sum = 0 ;
            for(int i=0; i<m; i++) sum += units[i][0] ;

            return sum ;
        }

        long long sum = 0 ;
        long long mini_f = 1e9, mini_s = 1e9  ;

        for(int i=0; i<m; i++)
        {
            long long sec_min = 1e9 ;
            long long fir_min = 1e9 ;

            for(int j=0; j<n; j++)
            {
                long long val = units[i][j] ;

                if(val < fir_min){
                    sec_min = fir_min ;
                    fir_min = val ;
                }
                else if(val < sec_min) sec_min = val ;
            }

            sum += sec_min ;
            mini_f = min(mini_f, fir_min) ;
            mini_s = min(mini_s, sec_min) ;
        }

        return sum - mini_s + mini_f ;

    }
};